classes/mail/transport/PHPMailTransport.php

```diff
@@ -0,0 +1,170 @@
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PHPMailTransport.php
+ *
+ * @brief Transport/adapter to send with PHPMailer
+ */
+
+namespace PKP\mail\transport;
+
+use APP\core\Application;
+use PHPMailer\PHPMailer\PHPMailer;
+use PKP\config\Config;
+use PKP\core\PKPString;
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Mime\RawMessage;
+use Exception;
+use ReflectionObject;
+
+class PHPMailTransport implements TransportInterface
+{
+    public const MAIL_WRAP = 76;
+
+    /**
+     * @inheritDoc
+     */
+    public function send(RawMessage $symfonyMessage, Envelope $envelope = null): ?SentMessage
+    {
+        if (!is_a($symfonyMessage, Email::class)) {
+            throw new Exception('Can\'t send raw message with phpmail');
+        }
+
+        $phpmailerMessage = $this->getPHPMailMessage($symfonyMessage);
+        try {
+            $success = $phpmailerMessage->Send();
+            if (!$success) {
+                error_log($phpmailerMessage->ErrorInfo);
+                return null;
+            }
+        } catch (Exception $e) {
+            error_log($phpmailerMessage->ErrorInfo);
+            return null;
+        }
+
+        return new SentMessage($symfonyMessage, $envelope ?? Envelope::create($symfonyMessage));
+    }
+
+    public function __toString(): string
+    {
+        return 'phpmail';
+    }
+
+    /**
+     * We won't send symfony message; transfer data to PHPMailer to send with mail()
+     */
+    protected function getPHPMailMessage(Email $symfonyMessage): PHPMailer
+    {
+        $mailer = new PHPMailer();
+        $mailer->isMail();
+        $mailer->IsHTML();
+        $mailer->Encoding = PHPMailer::ENCODING_BASE64;
+        $mailer->CharSet = $symfonyMessage->getHtmlCharset() ?? 'utf-8';
+        $mailer->XMailer = 'Public Knowledge Project Suite v3';
+        $mailer->WordWrap = static::MAIL_WRAP;
+
+        $request = Application::get()->getRequest();
+        $fromAddresses = $symfonyMessage->getFrom();
+        if (!empty($fromAddresses)) {
+            $f = current($fromAddresses);
+            if (Config::getVar('email', 'force_default_envelope_sender') && Config::getVar('email', 'default_envelope_sender') && Config::getVar('email', 'force_dmarc_compliant_from')) {
+                /* If a DMARC compliant RFC5322.From was requested we need to promote the original RFC5322.From into a Reply-to header
+                 * and then munge the RFC5322.From */
+                $alreadyExists = false;
+                foreach ($symfonyMessage->getReplyTo() as $r) {
+                    if ($r->getAddress() === $f->getAddress()) {
+                        $alreadyExists = true;
```

Comment:
I think you can add a break here, or perhaps use the https://laravel.com/docs/9.x/helpers#method-array-first

Reply:
Yep, look like it